src/index.rs

```diff
@@ -299,9 +331,12 @@ impl IndexTable {
 		let new_entry = Entry::new(address, partial_key, self.id.index_bits());
 		if let Some(i) = sub_index {
 			let entry = Entry::from_u64(u64::from_le_bytes(chunk[i * 8 .. i * 8 + 8].try_into().unwrap()));
+			// [Review] Can it be just debug_assert?
 			assert!(entry.key_material(self.id.index_bits()) == new_entry.key_material(self.id.index_bits()));
 			&mut chunk[i * 8 .. i * 8 + 8].copy_from_slice(&new_entry.as_u64().to_le_bytes());
 			log::trace!(target: "parity-db", "{}: Replaced at {}.{}: {}", self.id, chunk_index, i, new_entry.address(self.id.index_bits()));
+			// [Review] is changing the whole chunk necessary: we just need individual change over in
+			// memory log.
```

Comment:
It's just easier to implement a page overlay. If we move to k-ary search or some other form of sorted chunks, it will probably be necessary to overwrite the whole page anyway.